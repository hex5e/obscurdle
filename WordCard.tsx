import React from 'react';
import { View, Text, StyleSheet } from 'react-native';

export function WordCard({
  word,
  definition,
  reveal,
}: {
  word: string;
  definition: string;
  reveal?: boolean;
}) {
  return (
    <View style={styles.card}>
      <Text style={styles.word}>{word}</Text>
      {reveal && <Text style={styles.definition}>{definition}</Text>}
    </View>
  );
}

const styles = StyleSheet.create({
  card: {
    marginBottom: 20,
    padding: 24,
    backgroundColor: '#ffffff',
    borderRadius: 12,
    shadowColor: '#000',
    shadowOpacity: 0.1,
    shadowOffset: { width: 0, height: 4 },
    shadowRadius: 12,
    elevation: 3,
  },
  word: {
    fontSize: 28,
    marginBottom: 8,
  },
  definition: {
    fontSize: 18,
    lineHeight: 24,
    color: '#333',
  },
});
