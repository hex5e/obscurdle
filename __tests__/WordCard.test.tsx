import React from 'react';
import ReactDOMServer from 'react-dom/server';
import { WordCard } from '../WordCard';

jest.mock('react-native');

describe('WordCard', () => {
  test('renders word only when not revealed', () => {
    const html = ReactDOMServer.renderToStaticMarkup(
      <WordCard word="test" definition="A test" />
    );
    expect(html).toContain('test');
    expect(html).not.toContain('A test');
  });

  test('renders word and definition when revealed', () => {
    const html = ReactDOMServer.renderToStaticMarkup(
      <WordCard word="test" definition="A test" reveal />
    );
    expect(html).toContain('test');
    expect(html).toContain('A test');
  });
});
