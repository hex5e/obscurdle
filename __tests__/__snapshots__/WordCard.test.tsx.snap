// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`renders word and definition when revealed 1`] = `null`;

exports[`renders word only when not revealed 1`] = `null`;
